openra/handlers.py

```diff
@@ -223,89 +221,43 @@ def GetHash(self, filepath="", parser=settings.OPENRA_ROOT_PATH + list(reversed(
         if filepath == "":
             filepath = self.map_full_path_filename
 
-        os.chdir(parser + "/")
-
         os.chmod(filepath, 0o444)
 
-        command = 'mono --debug OpenRA.Utility.exe ra --map-hash ' + filepath
+        command = 'mono --debug %s ra --map-hash %s' % (os.path.join(parser, 'OpenRA.Utility.exe'), filepath)
         proc = Popen(command.split(), stdout=PIPE).communicate()
 
         os.chmod(filepath, 0o644)
 
         self.maphash = proc[0].decode().strip()
 
-        os.chdir(self.currentDirectory)
-
     def GenerateMinimap(self, game_mod, parser=settings.OPENRA_ROOT_PATH + list(reversed(list(settings.OPENRA_VERSIONS.values())))[0]):
```

Comment:
This is only used for MapFormat < 10, which was superseded in release-20160508.  Removing this plus other workarounds for old maps (and then, if needed, adding an error message when a too-old map format is uploaded) would be a good future PR topic.

Reply:
Yep, as above, this sounds like a good improvement, but should probably be looked into as a separate issue.